### PR TITLE
Fix `getViewPort` returning incorrect results

### PR DESCRIPTION
**Pull request by v-rob**
`IVideoDriver::getViewPort` has a bug where it only returns viewports set explicitly by `setViewPort`.  However, `OnResize` and `setRenderTarget(Ex)` also set the viewport since the drawing area has changed, but they only set it for OpenGL, not for `getViewPort`.  This PR fixes that.

Minetest currently doesn't do enough to run into any problems with this, but https://github.com/minetest/minetest/pull/10801 does, and this PR makes it work without a hacky workaround.

### CODE REVIEW

File: source/Irrlicht/COGLES2Driver.cpp [hunk 1]
```diff
@@ -2099,6 +2099,15 @@ COGLES2Driver::~COGLES2Driver()
 	}
 
 
+	void COGLES2Driver::setViewPortRaw(const core::rect<s32>& vp)
```
Comment (sfan5) [inline source/Irrlicht/COGLES2Driver.cpp:2102]: This should be a `core::dimension2d<u32>`, I don't see this being called with negative values anywhere

Comment (v-rob) [reply source/Irrlicht/COGLES2Driver.cpp]: (For that matter, `setViewPort` should be a core::rect<u32>, but who said Irrlicht made sense?)

Fixed.

File: source/Irrlicht/COGLES2Driver.cpp [hunk 2]
```diff
@@ -2099,6 +2099,15 @@ COGLES2Driver::~COGLES2Driver()
 	}
 
 
+	void COGLES2Driver::setViewPortRaw(u32 width, u32 height)
+	{
+		if (width > 0 && height > 0)
```
Comment (sfan5) [inline source/Irrlicht/COGLES2Driver.cpp:2104]: redundant, these are unsigned

File: source/Irrlicht/COGLES2Driver.cpp [hunk 3]
```diff
@@ -2252,7 +2261,7 @@ COGLES2Driver::~COGLES2Driver()
 	void COGLES2Driver::OnResize(const core::dimension2d<u32>& size)
 	{
 		CNullDriver::OnResize(size);
-		CacheHandler->setViewport(0, 0, size.Width, size.Height);
+		setViewPort(core::recti(0, 0, size.Width, size.Height));
```
Comment (sfan5) [inline source/Irrlicht/COGLES2Driver.cpp:2264]: changing `CNullDriver::OnResize(` to set `ViewPort` in all cases appears to work just fine, then you don't need to change the implementation in each driver

**Pull request merged**